### ISSUE TITLE
alloptions race conditions

### ISSUE THREAD

**Issue by soulseekah** [pressjitsu/pj-object-cache-red#18]
https://core.trac.wordpress.org/ticket/31245

Raised by @mihadan and probably @msigley when he said "the notorious 'alloptions' bug" in https://github.com/pressjitsu/pj-object-cache-red/pull/13#issuecomment-589294191

Handled in PR #6 for now, but needs a refresh.

**Comment by soulseekah** [comment]
Why didn't they use https://github.com/soulseekah/wp-lock 😸 ?

Are there any race conditions in our preloaded implementation?

**Comment by msigley** [comment]
Yes there are. The race condition occurs on sets to different keys on concerent requests. It occurs because all of the values are stored in one big blob instead of individual keys.

**Comment by msigley** [comment]
The fix for redis would be to replace the alloptions key with an mget.
